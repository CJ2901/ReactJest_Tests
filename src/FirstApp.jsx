import PropTypes from 'prop-types';

export const FirstApp = ( { title , subTitle, name } ) => {

    return (
        <>
            <h1 data-testid='test-title'> Hola, soy { title } </h1>
            {/* <code>{JSON.stringify(objetoMensaje)}</code> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }. Derechos reservados</p>
        </>
    )

}

FirstApp.propTypes = {
    subTitle: PropTypes.string,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Alexander Magallanes',
    subTitle: 'No hay subt√≠tulo en este episodio',
    title: 'el PoderMaximo',
}
