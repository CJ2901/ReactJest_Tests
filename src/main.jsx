import React from 'react';
import ReactDOM from 'react-dom/client'
import { Counterapp } from './Counterapp';
import { FirstApp } from './FirstApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title="The Batman" subTitle="The Dark Knight" /> */}
        <Counterapp value={10}/>
    </React.StrictMode>

);