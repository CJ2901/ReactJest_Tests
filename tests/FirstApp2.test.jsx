import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => { 
    
    const title = 'The Batman';
    const subTitle = 'The Dark Knight';

    test('should hacer match con el snapshot', () => { 
        
        const { container } = render( <FirstApp title={title} /> );      
        expect( container ).toMatchSnapshot();

    });
     
     test('debe de mostrar el mensaje del título', () => { 

        render( <FirstApp title={title} /> );
        expect( screen.getByText(`Hola, soy ${title}` ) ).toBeTruthy();
        // screen.debug();
        
    });
    
    test('should mostrar el título en un h1', () => { 
        
        render( <FirstApp title={title} /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(title);
        
    });

    test('should mostrar el subtítulo enviado por props', () => { 
        
        render( 
            <FirstApp 
                title={title}
                subTitle={subTitle}
            /> 
        );
        
        expect( screen.getAllByText(subTitle).length ).toBe(2);
        
    });

 })