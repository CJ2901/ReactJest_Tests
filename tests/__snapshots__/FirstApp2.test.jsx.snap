// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> should hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     Hola, soy 
    The Batman
     
  </h1>
  <p>
    No hay subtítulo en este episodio
  </p>
  <p>
    No hay subtítulo en este episodio
  </p>
  <p>
    Alexander Magallanes
    . Derechos reservados
  </p>
</div>
`;
